concepts/default-enarx-trust-process.md

```diff
@@ -0,0 +1,474 @@
+
+The Tenant Workload MUST never be unencrypted on the network (in transit)
+within the Enarx-managed process.
+
+The Tenant Workload MUST never be unencrypted in storage once it has been
+transmitted by the Enarx client agent.
+
+The Tenant Workload MUST never be unencrypted in RAM once it has been
+transmitted by the Enarx client agent.
+
+The Tenant Workload MUST never be available in unencrypted form on the host.
+
+### Other requirements
+The channel between the Enarx client agent and the Enarx host agent MAY be
+encrypted.
+
+The Enarx client agent SHOULD authenticate to the Enarx host agent.
+
+Integrity protection of the request SHOULD be applied.
+
+The protocol handshake and further steps SHOULD be encrypted.
+
+If integrity protection is applied, all detected integrity failures SHOULD
+be reported to the Enarx client agent via the Enarx host agent where possible.
+
+The integrity of the tenant's trust domain MUST always be given the
+highest priority.
+
+Protections for the tenant MUST always be prioritised above those of
+other parties such as the host.
+
+Confidentiality of the tenant's workload MUST be protected.
+
+Integrity of the tenant's workload MUST be protected.
+
+All protocol errors SHOULD be logged by the Enarx client agent.
+
+All attestation measurements MUST be cryptographically signed.
+
+All attestation measurements MUST be checked.
+
+All attestation measurements SHOULD be logged by the Enarx client agent.
+
+All Enarx client agent logs SHOULD be integrity protected.
+
+All attestation measurement failures MUST result in a cessation of
+the standard state progression.
+
+The cryptographic validity of the host CPU + CPU firmware MUST be
+checked in every attestation measurement check.
+
+All cryptographic validity check failures of the host CPU + CPU firmware
+MUST result in an attestation failure.
+
+The cessation of the standard state progression before transmission of
+the tenant workload MUST result in non-transmission of the workload from
+the Enarx client agent to any component on the host.
+
+All attestation measurement failures MUST be logged by the Enarx host
+agent.
+
+All attestation measurement failures SHOULD result in an error message
+being sent to the Orchestrator.
+
+The Enarx client agent MUST generate and include session tracking
+information in the initial communication with the Enarx host agent.
+
+The session tracking information MUST be included in all communications
+between the Enarx client agent and the Enarx host agent until the Keep
+is fully provisioned.
+
+The Tenant Workload MUST never be unencrypted on the network (in transit)
+within the Enarx-managed process.
+
+The Tenant Workload MUST never be unencrypted in storage once it has been
+transmitted by the Enarx client agent.
+
+The Tenant Workload MUST never be unencrypted in RAM once it has been
+transmitted by the Enarx client agent.
+
+The Tenant Workload MUST never be available in unencrypted form on the host.
```

Comment:
These are already in the general requirements section.

Reply:
Yes, I left them in just in case we wish to separate out the general requirements section to another document.  Need to tidy up once we decide.